pyproject.toml

```diff
@@ -10,6 +10,8 @@ select = [
     "F",
     # isort
     "I",
+    # numpy
+    "NPY",
```

Comment:
Deployment of the `examples/rapids-azureml-hpo/train_rapids.py` script failed like this:

```text
Traceback (most recent call last):
  File "/mnt/azureml/cr/j/525ffdb43cda47b1bd9386f0b02d17ae/exe/wd/train_rapids.py", line 172, in <module>
    main()
  File "/mnt/azureml/cr/j/525ffdb43cda47b1bd9386f0b02d17ae/exe/wd/train_rapids.py", line 65, in main
    mlflow.log_param("n_estimators", np.int(args.n_estimators))
                                     ^^^^^^
  File "/opt/conda/lib/python3.12/site-packages/numpy/__init__.py", line 394, in __getattr__
    raise AttributeError(__former_attrs__[attr])
AttributeError: module 'numpy' has no attribute 'int'.
`np.int` was a deprecated alias for the builtin `int`. To avoid this error in existing code, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
The aliases was originally deprecated in NumPy 1.20; for more details and guidance see the original release note at:
    https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations. Did you mean: 'inf'?
```

Because it's using things like `np.int()` that were removed in NumPy 2.0, and NumPy 2.x is making it into the environment.

Adding this `ruff` rules catches and auto-fixes such things.